src/ADAL.PCL/Internal/Telemetry/ApiEvent.cs

```diff
@@ -0,0 +1,155 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.IdentityModel.Clients.ActiveDirectory
+{
+    internal class ApiEvent : DefaultEvent
+    {
+        internal ApiEvent(Authenticator authenticator, UserInfo userinfo, string tenantId, string apiId)
+        {
+            Tenant = PlatformPlugin.CryptographyHelper.CreateSha256Hash(tenantId);
+            SetEvent(EventConstants.Tenant, Tenant);
+
+            if (userinfo != null)
+            {
+                Idp = userinfo.IdentityProvider;
+                SetEvent(EventConstants.Idp, Idp);
+
+                DisplayableId = PlatformPlugin.CryptographyHelper.CreateSha256Hash(userinfo.DisplayableId);
+                SetEvent(EventConstants.DisplayableId, DisplayableId);
+
+                UniqueId = PlatformPlugin.CryptographyHelper.CreateSha256Hash(userinfo.UniqueId);
+                SetEvent(EventConstants.UniqueId, UniqueId);
+            }
+
+            if (tenantId != null)
+            {
+                Authority = HashTenantIdFromAuthority(authenticator.Authority, tenantId);
+            }
+            else
+            {
+                Authority = authenticator.Authority;
+            }
+            SetEvent(EventConstants.Authority, Authority);
+
+            AuthorityType = authenticator.AuthorityType.ToString();
+            SetEvent(EventConstants.AuthorityType, AuthorityType);
+
+            IsDeprecated = false;
+            SetEvent(EventConstants.IsDeprecated, IsDeprecated.ToString());
+
+            SetEvent(EventConstants.ApiId, apiId);
+        }
+
+        internal string Tenant { get; set; }
+
+        internal string Idp { get; set; }
+
+        internal string Upn { get; set; }
+
+        internal string Authority { get; set; }
+
+        internal string AuthorityType { get; set; }
+
+        internal string ValidateAuthority { get; set; }
+
+        internal string DisplayableId { get; set; }
+
+        internal string IdentityProvider { get; set; }
+
+        internal string UniqueId { get; set; }
+
+        internal bool IsDeprecated { get; set; }
+
+        internal bool ExtendedExpires { get; set; }
+
+        private string HashTenantIdFromAuthority(string authority, string tenantId)
```

Comment:
this will not work if provided authority was "[https://login.microsoftonline.com/contoso.com](https://login.microsoftonline.com/contoso.com)". TenantID is a guid. we ONLY replace tenan name of it is "common".